components/telemetry-operator/internal/secret/secret_handling.go

```diff
@@ -0,0 +1,76 @@
+package secret
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	telemetryv1alpha1 "github.com/kyma-project/kyma/components/telemetry-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type SecretValidator struct {
```

Comment:
Since this does not only validate secrets, let's find a better name. How about `SecretHelper`?